src/main.js

```diff
@@ -31,6 +32,28 @@ function handleUnitSwitchClick(e) {
   updateWeather(weatherData, unitSystem);
 }
 
+async function handleDisplayWeatherBySearch() {
+  let searchText = document.getElementById("string").value;
+  let splitText = searchText.split(",");
+  let city = "";
+  let stateCode = "";
+  let countryCode = "";
+  let coords;
+  if (splitText.length < 3) {
+    city = splitText[0].trim();
+    countryCode = splitText[1].trim();
+    coords = await getLocationCoords(city, countryCode);
+  } else {
+    city = splitText[0].trim();
+    stateCode = splitText[1].trim();
+    countryCode = splitText[2].trim();
+    coords = await getLocationCoords(city, stateCode, countryCode);
+  }
+
+  weatherData = await getWeatherDetails(coords[0], coords[1]);
```

Comment:
I think you are overthinking this.

The user is not going to always insert `city, state, country` but sometimes only `city` or `state` perhaps.
If that happens you code is going to crash because `splitText[1]` is going to be undefined.

Even their example uses only city name.
```
http://api.openweathermap.org/geo/1.0/direct?q=London&limit=5&appid={API key}
```

The code should only make sure there is no extra whitespace in the input.

```
input = "   London " // => "London"
input = "London, GB " // => "London,GB"
```
and then the function `geolocation()` can accept only a single string..

Reply:
Ok, thanks i'll be changing them